STIPS/STIP-004.md

```diff
@@ -0,0 +1,730 @@
+
+## Open Questions
+
+1. Do we restrict usage of these modified issuance modules (which replace balance checks with collateralization checks) from third-party use?
+    - Why did we have the strict checks in the first place?
+        - To make sure the set is not undercollateralized 
+            - Eg. To stop tokens which collects fees on transfer from being a component in a SetToken
+        - This strict check achieved the above requirement, but at the same time it also prevented overcollateralization during issuance/redemption.
+    - What are the only requirements for our system to function well?
+        - We do not want Set to be undercollateralized
+        - We are fine with overcollateralization (discussed below)
+    - Thus, the proposed checks fulfill our requirements.
+        - So, we need not restrict usage of these modified issuance modules
+    
+2. Are we fine with Set being overcollaterlized?
+   * Yes, SetToken can resync the extra assets present to increase their default positions with help of modules
+      * ALM and CLM can resync positions for leveraged tokens
+      * Airdrop module can add extra assets as positions for other SetTokens
+   * Also, issuance/redemption generally would not lead to overcollaterlization of more than 1 wei
+      * Over-collateralization of 1 wei can be introduced to rounding errors
+      * For over collateralization greater than 1 wei, we would need the special case of a token which rewards holders on a transfer
+      * We can handle that case using Airdrop module
+
+3. Do we restrict these new modified issuance modules (which replace balance checks with collateralization checks) only to SetToken which hold aTokens? Or do we make these issuance modules the default going forward?
+    -  Current approach is leaning towards not restricting module usage but providing it as a potential different offering for manager's to choose when selecting the issuance module they want to use
+    - They can choose the default issuance modules with stricter checks if they are sure of not using any aToken or tokens which take fees on transfer as components in their SetToken
+    - But if they plan to use aTokens or tokens which take fees on transfer as SetToken components, then they can use the new issuance modules
+
+
+## Checkpoint 1
+
+**Reviewer**:
+
+## Proposed Architecture Changes
+### IssuanceUtils 
```

Comment:
Let's be a little more descriptive with this name...`IssuanceCollateralizationUtils` or something like that